src/NetworkName.js

```diff
@@ -1,60 +1,136 @@
-/**
- * @typedef {import("./client/Client.js").NetworkName} ClientNetworkName
- */
+import LedgerId from "./LedgerId.js";
 
-/**
- * @typedef {object} NetworkNameType
- * @property {ClientNetworkName} Mainnet
- * @property {ClientNetworkName} Testnet
- * @property {ClientNetworkName} Previewnet
- */
+const DEFAULT_ERROR = "Default case reached for: ";
 
 /**
- * @type {NetworkNameType}
+ * @private
  */
-const NetworkName = {
-    Mainnet: "mainnet",
-    Testnet: "testnet",
-    Previewnet: "previewnet",
-};
+export default class NetworkName {
+    /**
+     * @hideconstructor
+     * @internal
+     * @param {string} networkName
+     */
+    constructor(networkName) {
```

Comment:
Might be better to require the parameter to be of type `number` where 0 = mainnet, 1 = testnet, 2 = previewnet, and all else is treated as other.